neptune_lightgbm/impl/__init__.py

```diff
@@ -79,7 +94,79 @@ def __init__(self, run: 'neptune.Run', base_namespace=''):
         self._run['source_code/integrations/neptune-lightgbm'] = __version__
 
     def __call__(self, env):
-        for name, loss_name, loss_value, _ in env.evaluation_result_list:
-            channel_name = '{}{}_{}'.format(self._base_namespace, name, loss_name)
+        # TODO:
+        # (data_name, eval_name, val, is_higher_better): `lightgbm.basic.Booster.__inner_eval`
+        # ('cv_agg', k, np.mean(v), metric_type[k], np.std(v)): `lightgbm.engine._agg_cv_result`
+        for name, loss_name, loss_value, *_ in env.evaluation_result_list:
+            channel_name = 'foo/{}/foo2/{}_{}'.format(self._base_namespace, name, loss_name)
             self._run[channel_name].log(loss_value, step=env.iteration)
 
+
+def create_booster_summary(
+        booster: Union[lgb.Booster, lgb.sklearn.LGBMModel],
+        log_importances: bool = True,
+        max_num_features: int = 10,
+        list_trees: list = None,
+        log_trees_as_dataframe: bool = True,  # works only for lgb.Booster
+        log_pickled_booster: bool = True,
+        log_trees: bool = False,  # requires graphviz
+        log_confusion_matrix: bool = False,  # requires scikit-plot
+        y_true: np.ndarray = None,
+        y_pred: np.ndarray = None,
+):
+    results_dict = {}
+    visuals_path = "visualizations/"
+    if log_importances:
+        split_plot = lgb.plot_importance(
+            booster,
+            importance_type="split",
+            title="Feature importance (split)",
+            max_num_features=max_num_features
+        )
+        gain_plot = lgb.plot_importance(
+            booster,
+            importance_type="gain",
+            title="Feature importance (gain)",
+            max_num_features=max_num_features
+        )
+        results_dict["{}feature_importances/split".format(visuals_path)] \
+            = neptune.types.File.as_image(split_plot.figure)
+        results_dict["{}feature_importances/gain".format(visuals_path)] \
+            = neptune.types.File.as_image(gain_plot.figure)
+
+    # TODO: Your file is larger than 15MB. Neptune supports logging files in-memory objects smaller than 15MB.
+    #       Resize or increase compression of this object
```

Comment:
For sample code I've tried we've failed to upload some data.

Do we want to fix it somehow, or leave it as it is?

Reply:
We want to inform user about:
* fact that upload was unsuccessful,
* what was not logged (tree or model, or sth. else)
* inform that it's a warning not error.

Reply:
Done:
Additional warning is displayed: `'trees_as_dataframe' wasn't logged. Probably generated dataframe was to large`.
When dataframe is non empty, but no data was uploaded to server.

I can't intercept additional `Your file is larger than 15MB....` message, because it's logged in `neptune-client` code.

Reply:
Looks good, maybe we can rephrase it a bit:

```
'trees_as_dataframe' wasn't logged.
Generated DataFrame was too large (>15MB).
Try to log it, by first saving the file to disk and then logging it by using: run["trees_as_dataframe"].upload('path to file').
```